flask_static_digest/__init__.py

```diff
@@ -26,28 +26,29 @@ def init_app(self, app):
         app.config.setdefault("FLASK_STATIC_DIGEST_HOST_URL", None)
 
         self.host_url = app.config.get("FLASK_STATIC_DIGEST_HOST_URL")
-        self.static_url_path = app.static_url_path
 
-        self.manifest_path = os.path.join(app.static_folder,
-                                          "cache_manifest.json")
-        self.has_manifest = os.path.exists(self.manifest_path)
+        self.manifests = {}
 
-        self.manifest = {}
+        def ingestScaffold(endpoint, scaffold):
```

Comment:
As for naming, where did the phrase "scaffold" come from?

Reply:
A Flask app inherits Scaffold: https://github.com/pallets/flask/blob/main/src/flask/app.py#L105
A Flask blueprint inherits Scaffold: https://github.com/pallets/flask/blob/main/src/flask/blueprints.py#L117

The Scaffold class is responsible for the static folder polymorphism between apps and blueprints.
https://github.com/pallets/flask/blob/2f67e0fe4aca8a8d6998b3022150f1e6d6738214/src/flask/scaffold.py#L54

Reply:
Thanks for the clarification. If that's how it's defined at the Flask level I'm ok with the name scaffold. I'll make a follow up comment on suggestions about the function's logic.

Reply:
This comment applies to the following code:

```py
        def ingestScaffold(endpoint, scaffold):
            if not scaffold.has_static_folder:
                return

            manifest_path = os.path.join(scaffold.static_folder,
                                         "cache_manifest.json")
            has_manifest = os.path.exists(manifest_path)

            if has_manifest:
                manifest = self._load_manifest(scaffold, manifest_path)
                self.manifests[endpoint] = manifest

        ingestScaffold("static", app)
        for endpoint, scaffold in app.blueprints.items():
            ingestScaffold(f"{endpoint}.static", scaffold)
```

Can we:

- Rename `ingestScaffold` to `load_manifests`

Also are we going to get into trouble by hard coding `("static", app)` and `f"{endpoint}.static"`. Are these always guaranteed to be named `static` no matter what type of customization an end user will do using Flask's public interface?

Reply:
Also what do you think about renaming `for endpoint, scaffold in app.blueprints.items():` to `for endpoint, blueprint in app.blueprints.items():`. I think it's a little more clear since we're looping over blueprint items. The blueprint is the item in the loop. That would mean changing the line below it to reference `blueprint` too.

Reply:
I'm in favor of renaming `ingestScaffold` to `load_manifest`

I'll begrudgingly rename the `scaffold` argument to `blueprint`.
It's just that `def load_manifest(endpoint, blueprint):` is going to feel awkward when the argument isn't a blueprint.
i.e. `load_manifest("static", app)`.

Static endpoints are hardcoded as "static" here:
https://github.com/pallets/flask/blob/main/src/flask/app.py#L513
https://github.com/pallets/flask/blob/main/src/flask/blueprints.py#L322

I'm for sure going to review how blueprints are iterated, along with the CLI

Reply:
Ok thanks, sounds good.

Let's roll with:

- `load_manifest` and keep its argument named as `scaffold`
- Keep the hard coded `static`

Reply:
Where do you stand on https://github.com/nickjj/flask-static-digest/pull/33#discussion_r1133085554? This one feels like it would always be a blueprint, in which case we could change its name?

Reply:
[0501677 renaming scaffold and ingestScaffold, reworked load_manifest](https://github.com/nickjj/flask-static-digest/pull/33/commits/05016779a5593dd065231c662fd6040c2fb6e09e)

Turns out there was already a function named `_load_manifest`. I ended up merging `ingestScaffold` with `_load_manifest` and reworked the code a little bit as I went. The file IO is going to be more robust now and less likely to crash the server.